pkg/routeagent_driver/handlers/ovn/connection.go

```diff
@@ -0,0 +1,195 @@
+	nbdbModel, err := nbdb.FullDatabaseModel()
+	if err != nil {
+		return errors.Wrap(err, "error getting OVN NBDB database model")
+	}
+
+	c.nbdb, err = c.createLibovsdbClient(getOVNNBDBAddress(), tlsConfig, nbdbModel)
+	if err != nil {
+		return errors.Wrap(err, "error creating NBDB connection")
+	}
+
+	return nil
+}
+
+func getOVNTLSConfig(pkFile, certFile, caFile string) (*tls.Config, error) {
+	cert, err := tls.LoadX509KeyPair(certFile, pkFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failure loading ovn certificates")
+	}
+
+	rootCAs := x509.NewCertPool()
+
+	data, err := os.ReadFile(caFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "failure loading OVNDB ca bundle")
+	}
+
+	rootCAs.AppendCertsFromPEM(data)
+
+	return &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		RootCAs:      rootCAs,
+		ServerName:   "ovn",
+		MinVersion:   tls.VersionTLS12,
+	}, nil
+}
+
+func (c *ConnectionHandler) createLibovsdbClient(dbAddress string, tlsConfig *tls.Config,
+	dbModel model.ClientDBModel,
+) (libovsdbclient.Client, error) {
+	options := []libovsdbclient.Option{
+		// Reading and parsing the DB after reconnect at scale can (unsurprisingly)
+		// take longer than a normal ovsdb operation. Give it a bit more time so
+		// we don't time out and enter a reconnect loop.
+		libovsdbclient.WithReconnect(OVSDBTimeout, &backoff.ZeroBackOff{}),
+		libovsdbclient.WithLogger(&logger.Logger),
+	}
+
+	if strings.HasPrefix(dbAddress, "IC:") {
+		node, err := nodeutil.GetLocalNode(c.k8sClientset)
+		if err != nil {
+			return nil, errors.Wrap(err, "error getting the node")
+		}
+
+		annotations := node.GetAnnotations()
+
+		zoneName, ok := annotations[constants.OvnZoneAnnotation]
+		if !ok {
+			return nil, errors.Wrap(err, "error getting the zone name")
+		}
+
+		zoneDBAddress := getICDBAddress(dbAddress, zoneName)
+		options = append(options, libovsdbclient.WithEndpoint(zoneDBAddress))
+	} else {
+		options = append(options, libovsdbclient.WithEndpoint(dbAddress))
+	}
+
+	if tlsConfig != nil {
+		options = append(options, libovsdbclient.WithTLSConfig(tlsConfig))
+	}
+
+	client, err := libovsdbclient.NewOVSDBClient(dbModel, options...)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating ovsdbClient")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), OVSDBTimeout)
+	defer cancel()
+
+	err = client.Connect(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "error connecting to ovsdb")
+	}
+
+	if dbModel.Name() == "OVN_Northbound" {
+		_, err = client.MonitorAll(ctx)
+		if err != nil {
+			client.Close()
+			return nil, errors.Wrap(err, "error setting OVN NBDB client to monitor-all")
+		}
+	} else {
+		// Only Monitor Required SBDB tables to reduce memory overhead
+		_, err = client.Monitor(ctx,
+			client.NewMonitor(
+				libovsdbclient.WithTable(&sbdb.Chassis{}),
+			),
+		)
+		if err != nil {
+			client.Close()
+			return nil, errors.Wrap(err, "error monitoring chassis table in OVN SBDB")
+		}
+	}
```

Comment:
We can consolidate error handling a bit.

```suggestion
	err = errors.Wrap(err, "error connecting to ovsdb")

	if err != nil {
		if dbModel.Name() == "OVN_Northbound" {
			_, err = client.MonitorAll(ctx)
			err = errors.Wrap(err, "error setting OVN NBDB client to monitor-all")
		} else {
			// Only Monitor Required SBDB tables to reduce memory overhead
			_, err = client.Monitor(ctx,
				client.NewMonitor(
					libovsdbclient.WithTable(&sbdb.Chassis{}),
				),
			)
			err = errors.Wrap(err, "error monitoring chassis table in OVN SBDB")
		}
	}

	if err != nil {
		client.Close()
		return nil, err
	}
```

Reply:
Done.